userspace/libsinsp/container.cpp

```diff
@@ -1293,6 +1359,67 @@ string sinsp_container_manager::get_container_name(sinsp_threadinfo* tinfo)
 	return res;
 }
 
+void sinsp_container_manager::identify_healthcheck(sinsp_threadinfo *tinfo)
+{
+	// This thread is a part of a container healthcheck if its
+	// parent thread is part of a health check.
+	sinsp_threadinfo* ptinfo = tinfo->get_parent_thread();
+
+	if(ptinfo && ptinfo->m_is_container_healthcheck)
+	{
+		tinfo->m_is_container_healthcheck = true;
+		return;
+	}
+
+	sinsp_container_info *cinfo = get_container(tinfo->m_container_id);
+
+	if(!cinfo)
+	{
+		return;
```

Comment:
maybe a debug log?

Reply:
See above comment...
e.g. shouldn't we set tinfo->m_is_container_healthcheck to false in this case?

Reply:
Not having cinfo will be pretty common, as it will occur for any process that doesn't run inside a container.

I'll set m_is_container_healthcheck to false at the top of the function, although it should already be false from the sinsp_threadinfo's constructor.

Reply:
Actually, it turns out there are some cases where a process like `sh -c /bin/healthcheck` will exec directly to `/bin/healthcheck` without a fork. This means that the threadinfo initially meets the conditions to be labeled a healthcheck, but after the exec won't actually have a parent that is a healthcheck, even though it really is still associated with the initial healthcheck process. In this case, we should not change the state of m_is_container_healthcheck in `sinsp_container_manager::identify_healthcheck`, but only set it to true if it meets the conditions.